src/affine_decomposition.jl

```diff
@@ -60,16 +48,59 @@ Perform the compression of an [`AffineDecomposition`](@ref)
 corresponding to ``o = B^\\dagger O B``.
 """
 function compress(ad::AffineDecomposition, basis::RBasis)
-    AffineDecomposition(compress.(ad.terms, Ref(basis)), ad.coefficient_map)
+    matrixel = compress.(ad.terms, Ref(basis.snapshots))  # ⟨ψᵢ|O|ψⱼ⟩
+    rbterms = map(m -> basis.vectors' * m * basis.vectors, matrixel)  # transform to RB
+    (; rb=AffineDecomposition(rbterms, ad.coefficient_map),
+     raw=AffineDecomposition(matrixel, ad.coefficient_map))
 end
+
+"""
+    compress(ad::AffineDecomposition{<:Matrix,<:Function},
+             basis::RBasis; symmetric_terms=false)
+
+Perform compression for an [`AffineDecomposition`](@ref) with terms
+with two indices (double-sum observables), including an option to
+exploit the possible symmetry of terms ``O_{r,r'} = O_{r',r}``,
+such that only the necessary compressions are computed.
+"""
+function compress(ad::AffineDecomposition{<:Matrix,<:Function},
+                  basis::RBasis; symmetric_terms=false)
+    # TODO: replace symmetric_terms by some generalized Symmetric type in the long run
+    if symmetric_terms
+        ⪒(a, b) = (size(ad.terms, 1) > size(ad.terms, 2)) ? ≥(a, b) : ≤(a, b)
+        matrixel = map(CartesianIndices(ad.terms)) do idx
+            if ⪒(first(idx.I), last(idx.I))  # Upper/lower triangle
+                return compress(ad.terms[idx], basis.snapshots)
+            end
+        end
+        for idx in findall(x -> isnothing(x), matrixel)
+            matrixel[idx] = matrixel[last(idx.I), first(idx.I)]
+        end  # Use "symmetry" to set transposed elements
+        matrixel = promote_type.(matrixel)  # Promote to common floating-point type
+    else
+        matrixel = compress.(ad.terms, Ref(basis.snapshots))
```

Comment:
Could this special-casing not be included above?

Reply:
Hmm I guess I could include this in the `findall(isnothing(x), matrixel)` for-loop, such that the `nothing` elements get explicitly computed when there is no redundancy.

Reply:
add this to the issue above.